xwiki-rendering-macros/xwiki-rendering-macro-figure/src/main/java/org/xwiki/rendering/internal/macro/figure/FigureCaptionMacro.java

```diff
@@ -89,15 +89,17 @@ public boolean supportsInlineMode()
 
         // If we're not inside a FigureBlock then don't do anything.
         Block parent = context.getCurrentMacroBlock().getParent();
-        if (parent instanceof FigureBlock) {
+        if (parent instanceof FigureBlock
+            || (parent instanceof MetaDataBlock && parent.getParent() instanceof FigureBlock))
```

Comment:
Small Note (not needed to accept the PR): I think we have some code elsewhere that also skips some blocks. Would be good to introduce some common API to decide if a parent block is of a given type, and skips blocks such as MetaDataBlock or MacroMarkerBlock.

I can remember code in the LaTeX repo at least but I'm sure there's more elsewhere. One idea is to provide a `getXXXParent()` method that returns an effective parent.

https://github.com/xwiki-contrib/latex/blob/master/latex-syntax/src/main/java/org/xwiki/contrib/latex/internal/DefaultLaTeXTool.java#L149

Needed here too: https://github.com/xwiki-contrib/macro-numbered-references/blob/f361080ce569ecfe280516908fd7abd7b03837dc/src/main/java/org/xwiki/contrib/numberedreferences/internal/NumberedFiguresTransformation.java#L196

Anyway, just a thought.